### PR TITLE
create release procedure

### PR DESCRIPTION
**Pull request by jagedn**
new GH action to run and publish a release

Once we're ready to publish a release we need to follow these steps:

- set and commit the new version in gradle.properties
- tag the source with the vx.y.z pattern

if all goes well the action will create a new release `x.y.z` , attach the artifacts and creates a changelog

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+name: release
+on:
+  push:
+    tags:
```
Comment (abhi18av) [inline .github/workflows/release.yml:4]: Hi @jagedn , 

So basically we'll just tag the plugin locally and then push the plugin to trigger the build process of artifacts right?

Comment (jagedn) [reply .github/workflows/release.yml:4]: exactly , this is the idea

and not only locally, also you can use the web interface

Comment (abhi18av) [reply .github/workflows/release.yml:4]: Ah, gotcha - then let's roll! 🛼

**Review by abhi18av — APPROVED** [review]
Looks good to merge and let's trigger a build for v0.1.1 once this is in :)

**Comment by jagedn** [comment]
in fact we can try it now , we need only to delete current release and create the tag 0.1.0 (in case you want to give a try)

**Comment by abhi18av** [comment]
Ah, I just tried here https://github.com/nextflow-io/nf-nomad/actions/runs/9761642912/usage 

Not sure why it is getting skipped 🤔

**Comment by jagedn** [comment]
my fault, it had some (resolved) bugs 

First release done: https://github.com/nextflow-io/nf-nomad/releases/tag/v0.1.1

**Comment by jagedn** [comment]
(still minor issues generating the final url but I'm on it)

**Comment by abhi18av** [comment]
No worries, Jorge - I made a release here with the corresponding version https://github.com/nextflow-io/nf-nomad/releases/tag/v0.1.1 


The generated assets are also showing up in the tagged releases and the release notes which are generated from that 🎉 


Adding a PR to `plugins` repo to add this new version now

**Comment by jagedn** [comment]
hahaha and I've deleted 

let me finish the issues with the GH action so we'll publish v0.1.1 using it

**Comment by abhi18av** [comment]
Oki doki 👍

**Comment by jagedn** [comment]
So, release GH action is working and release 0.1.1 was generated (and tested) 

Now you can grab the `json` attached in the release and create a PR in the nextflow plugin directory 

https://github.com/nextflow-io/nf-nomad/releases/tag/0.1.1

**Comment by abhi18av** [comment]
Thanks mate, as per your confirmation I'll publish the add the plugin to the index 

image

**Pull request merged**